### PR TITLE
Feedback request for Practical 5

### PR DESCRIPTION
**Pull request by jonathangrech**
@stevepons please check code efficiency and general format.  Cheers.

### CODE REVIEW

File: Prac05/countWords.py [hunk 1]
```diff
@@ -9,6 +9,16 @@
     else:
         word_dict[word] = 1
 
+word_lengths = []
```
Comment (ghost) [inline Prac05/countWords.py:12]: All looks good, no changes to suggest.

File: Prac05/colourNames.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                "PaleTurquoise": "#afeeee", "SaddleBrown": "#8b4513", "SteelBlue": "#4682b4"}
+
+for colour in COLOUR_CODES:
+    print("{:<20} is {}".format(colour, COLOUR_CODES[colour]))
+
+colour = input("Enter colour name: ")
+while colour != "":
+    if colour in COLOUR_CODES:
+        print(colour, "is", COLOUR_CODES[colour])
+        break
```
Comment (lindsaymarkward) [inline Prac05/colourNames.py:17]: Why are you using break here?

File: Prac05/colourNames.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    else:
+        print("Invalid colour name")
+        break
+state = input("Enter colour name: ")
```
Comment (lindsaymarkward) [inline Prac05/colourNames.py:21]: What's this? state?
It needs to be colour, and be inside the loop.

File: Prac05/countWords.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        word_dict[word] += 1
+    else:
+        word_dict[word] = 1
+
+word_lengths = []
+words = sorted(list(word_dict.keys()))
+for word in words:
+    word_lengths.append(len(word))
+
+width = max(word_lengths)
```
Comment (lindsaymarkward) [inline Prac05/countWords.py:17]: Try using a one-line list comprehension (generator) to calculate the maximum length.

File: Prac05/countWords.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+word_counts = list(word_dict.items())
+word_counts = sorted(word_counts, key=lambda x: x[0])
```
Comment (lindsaymarkward) [inline Prac05/countWords.py:20]: You could just get the keys and sort those; easier and faster.

File: Prac05/stateNames.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+state = input("Enter short state: ").upper()
+while state != "":
+    if state in STATE_NAMES:
+        print(state, "is", STATE_NAMES[state])
+        break
+    else:
+        print("Invalid short state")
+        break
+state = input("Enter short state: ")
```
Comment (lindsaymarkward) [inline Prac05/stateNames.py:20]: Same as comments on the colour program... Did you test this?

**Pull request merged**